runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/PuzzleSolverOverlay.java

```diff
@@ -137,6 +139,27 @@ public Dimension render(Graphics2D graphics)
 			}
 			else
 			{
+				if (config.displayNumbers()) {
+					for (int id : itemIds)
+					{
+						if (id != BLANK_TILE_VALUE)
```

Comment:
breaking this into a smaller method would allow you to inverse this if and return early